File: benchmarkDS/CoT/CLAUDE_claude-sonnet-4-5-20250929/Lse124.py

import sys
import re

def main():
    # Check if exactly one argument is provided
    if len(sys.argv) != 2:
        print("Error: Exactly one argument is required.", file=sys.stderr)
        sys.exit(1)
    
    # Get the argument from command line
    input_arg = sys.argv[1]
    
    # Print the argument
    print(f"Original: {input_arg}")
    
    # Trim trailing whitespace
    input_arg = input_arg.rstrip()
    
    # Print the trimmed argument
    print(f"Trimmed: {input_arg}")
    
    # Exit the program
    sys.exit(0)

if __name__ == "__main__":
    main()

File: benchmarkDS/ZeroShot/CLAUDE_claude-sonnet-4-5-20250929/Lse124.c

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void processInput(int argc, char* argv[]) {
    // Check if exactly one argument is provided
    if (argc != 2) {
        fprintf(stderr, "Error: Exactly one argument required\\n");
        exit(1);
    }
    
    // Get the argument from command line
    char* input = argv[1];
    
    // Print the argument
    printf("Original input: %s\\n", input);
    
    // Trim trailing whitespace
    int len = strlen(input);
    while (len > 0 && (input[len-1] == ' ' || input[len-1] == '\\t' || 
                       input[len-1] == '\\n' || input[len-1] == '\\r')) {
        input[len-1] = '\\0';
        len--;
    }
    
    // Print the argument again
    printf("Trimmed input: %s\\n", input);
}

int main() {
    // Test cases
    char* testCases[][2] = {
        {"prog", "Hello"},
        {"prog", "World   "},
        {"prog", "Test\\t\\t"},
        {"prog", "NoWhitespace"},
        {"prog", "Multiple Words  "}
    };
    
    for (int i = 0; i < 5; i++) {
        printf("Test Case %d:\\n", i + 1);
        char input_copy[100];
        strcpy(input_copy, testCases[i][1]);
        char* argv[2] = {testCases[i][0], input_copy};
        processInput(2, argv);
        printf("\\n");
    }
    
    return 0;
}
